packages/form-validations/src/utils.ts

```diff
@@ -0,0 +1,27 @@
+export const findAncestor = (
+  element: HTMLElement | null | undefined,
+  selector: string,
+): HTMLElement | null | undefined => {
+  // from bottom up find the element with the className
+  // eslint-disable-next-line no-cond-assign, no-param-reassign
+  while ((element = element?.parentElement) && !element.matches(selector));
+
+  return element;
+};
+
+export const tmpl = (message: string, params: Record<string, string>) =>
```

Comment:
What exactly should I imagine under this name `tmpl` ?

Reply:
Actually, it supports simple templating in messages. I have chosen the better naming.